templates/server/src/main/resources/META-INF/archetype-post-generate.groovy

```diff
@@ -0,0 +1,30 @@
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.nio.file.LinkOption
+
+// the path where the project got generated
+Path projectPath = Paths.get(request.outputDirectory, request.artifactId)
+
+// the properties available to the archetype
+Properties properties = request.properties
+String batch = properties.get("batch")
+
+if (batch != "batch") {
+  // delete folder batch recursively
+  Path rootPath = projectPath.resolve("batch")
+  deleteDirectoryRecursion(rootPath)
+}
+
+void deleteDirectoryRecursion(Path path) {
+  if (Files.isDirectory(path, LinkOption.NOFOLLOW_LINKS)) {
+    try {
+      for (Path entry : Files.newDirectoryStream(path)) {
+        deleteDirectoryRecursion(entry);
+      }
+    } catch(IOException e) {
+      e.printStackTrace()
```

Comment:
surely an exotic situation but we should avoid using `printStackTrace()`.
IMHO simply wrap the IOException into a new RuntimeException and throw that one to abort the process in case of unexepcted I/O error.